src/Microsoft.TemplateEngine.Cli/HelpAndUsage/InvalidParameterInfo.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.TemplateEngine.Cli.HelpAndUsage
+{
+    /// <summary>
+    /// The class represents the information about the invalid template parameter used when executing the command
+    /// </summary>
+    internal class InvalidParameterInfo
+    {
+        internal enum Kind
+        {
+            InvalidParameterName,
+            InvalidParameterValue,
+            InvalidDefaultValue,
+            AmbiguousParameterValue
+        }
+
+        internal InvalidParameterInfo(Kind kind, string inputFormat, string specifiedValue, string canonical)
+        {
+            ErrorKind = kind;
+            InputFormat = inputFormat;
+            SpecifiedValue = specifiedValue;
+            Canonical = canonical;
+        }
+
+        /// <summary>
+        /// the option used in CLI for parameter
+        /// </summary>
+        internal string InputFormat { get; }
+        /// <summary>
+        /// The value specified for the parameter in CLI
+        /// </summary>
+        internal string SpecifiedValue { get; }
+        /// <summary>
+        /// The canonical name for the option
+        /// </summary>
+        internal string Canonical { get; }
+        /// <summary>
+        /// - <see cref="Kind.InvalidParameterName"/> - the option name is invalid <br />
+        /// - <see cref="Kind.InvalidParameterValue"/> - the value is invalid <br />
+        /// - <see cref="Kind.InvalidDefaultValue"/> - the default name is invalid <br />
+        /// - <see cref="Kind.AmbiguousParameterValue"/> - the value provided leads to ambiguous choice (for choice parameters only) <br />
```

Comment:
Maybe put this documentation of `enum Kind` fields on fields themself instead of here?